app/src/main/java/com/infomaniak/swisstransfer/ui/screen/newtransfer/upload/UploadProgressScreen.kt

```diff
@@ -17,10 +17,236 @@
+                    style = SwissTransferTheme.typography.specificLight18,
+                    textAlign = TextAlign.Center,
+                )
+
+                WeightOneSpacer(minHeight = Margin.Medium)
+
+                Image(imageVector = adScreenType.illustration.image(), contentDescription = null)
+
+                WeightOneSpacer(minHeight = Margin.Medium)
+            }
+
+            Spacer(modifier = Modifier.height(Margin.Medium))
+            Text(stringResource(R.string.uploadSuccessTransferInProgress))
+            Progress(uploadedSizeInBytes, totalSizeInBytes)
+            Spacer(modifier = Modifier.height(Margin.Huge)) // TODO
+        }
+
+        if (showBottomSheet) CancelUploadBottomSheet(onCancel = onCancel, closeButtonSheet = { showBottomSheet = false })
+    }
+}
+
+@Composable
+private fun CancelUploadBottomSheet(onCancel: () -> Unit, closeButtonSheet: () -> Unit) {
+    SwissTransferBottomSheet(
+        titleRes = R.string.uploadCancelConfirmBottomSheetTitle,
+        imageVector = AppIllus.RedCrossPaperPlanes.image(),
+        topButton = {
+            LargeButton(
+                titleRes = R.string.uploadCancelConfirmBottomSheetCancel,
+                modifier = it,
+                style = ButtonType.ERROR,
+                onClick = onCancel,
+            )
+        },
+        bottomButton = {
+            LargeButton(
+                titleRes = R.string.uploadCancelConfirmBottomSheetClose,
+                modifier = it,
+                style = ButtonType.TERTIARY,
+                onClick = closeButtonSheet,
+            )
+        },
+        onDismissRequest = closeButtonSheet,
+    )
+}
+
+@Composable
+private fun ColumnScope.WeightOneSpacer(minHeight: Dp) {
+    Spacer(modifier = Modifier.height(minHeight))
+    Spacer(modifier = Modifier.weight(1f))
+}
+
+@Composable
+fun Progress(uploadedSizeInBytes: () -> Long, totalSizeInBytes: () -> Long) {
+    Row {
+        Percentage(uploadedSizeInBytes, totalSizeInBytes)
+        Text(" - ")
+        UploadedSize(uploadedSizeInBytes)
+
+        TotalSize(totalSizeInBytes)
+    }
+}
+
+@Composable
+private fun Percentage(uploadedSizeInBytes: () -> Long, totalSizeInBytes: () -> Long) {
+    val percentageNoDecimals by remember {
+        derivedStateOf {
+            val percentage = (uploadedSizeInBytes().toFloat() / totalSizeInBytes())
+            String.format(Locale.getDefault(), "%d", (percentage * 100).toInt())
+        }
+    }
+
+    Text("$percentageNoDecimals%")
+}
+
+@Composable
+private fun UploadedSize(uploadedSizeInBytes: () -> Long) {
+    val context = LocalContext.current
+    val humanReadableSize by remember {
+        derivedStateOf { HumanReadableSizeUtils.getHumanReadableSize(context, uploadedSizeInBytes()) }
+    }
+
+    Text(humanReadableSize)
+}
+
+@Composable
+private fun TotalSize(totalSizeInBytes: () -> Long) {
+    val context = LocalContext.current
+    val humanReadableTotalSize by remember {
```

Comment:
I think its better to avoid writing to the saveStateHandle with rememberSavable when we have access to the source we use to compute the humanReadableTotalSize